File: src/main/java/io/github/oxnz/Ingrid/dts/mq/TxEvent.java
package io.github.oxnz.Ingrid.dts.mq;

import java.io.Serializable;

public class TxEvent implements Serializable {
    final long id;

    public TxEvent(long id) {
        this.id = id;
    }

    public long getId() {
        return id;
    }
}

File: src/main/scala/io/github/oxnz/Ingrid/tx/TxEvent.scala
package io.github.oxnz.Ingrid.tx

import java.io.Serializable

class TxEvent(val id: Long) extends Serializable {
  override def toString = s"TxEvent(id=$id)"
}

object TxEvent {
  def apply(id: Long): TxEvent = new TxEvent(id)
  def unapply(arg: TxEvent): Long = arg.id
}
